src/backend/base/langflow/api/v1/flows.py

```diff
@@ -121,64 +123,80 @@ def create_flow(
+        remove_example_flows (bool, optional): Whether to remove example flows. Defaults to False.
     Returns:
-        List[Dict]: A list of flows in JSON format.
+        Union[list[FlowRead], Page[FlowRead]]: A list of flows or a paginated response containing the list of flows.
     """
 
     try:
         auth_settings = settings_service.auth_settings
+
+        default_folder = session.exec(select(Folder).where(Folder.name == DEFAULT_FOLDER_NAME)).first()
+        default_folder_id = default_folder.id if default_folder else None
+
+        starter_folder = session.exec(select(Folder).where(Folder.name == STARTER_FOLDER_NAME)).first()
+        starter_folder_id = starter_folder.id if starter_folder else None
+
+        if not folder_id:
+            folder_id = default_folder_id
+
         if auth_settings.AUTO_LOGIN:
             stmt = select(Flow).where(
                 (Flow.user_id == None) | (Flow.user_id == current_user.id)  # noqa: E711
             )
-            if components_only:
-                stmt = stmt.where(Flow.is_component == True)  # noqa: E712
-            flows = session.exec(stmt).all()
-
         else:
-            flows = current_user.flows
+            stmt = select(Flow).where(Flow.user_id == current_user.id)
+
+        if remove_example_flows:
+            stmt = stmt.where(Flow.folder_id != starter_folder_id)
 
-        flows = validate_is_component(flows)
         if components_only:
-            flows = [flow for flow in flows if flow.is_component]
-        flow_ids = [flow.id for flow in flows]
-        # with the session get the flows that DO NOT have a user_id
-        folder = session.exec(select(Folder).where(Folder.name == STARTER_FOLDER_NAME)).first()
-
-        if not remove_example_flows and not components_only:
-            try:
-                example_flows = folder.flows if folder else []
-                for example_flow in example_flows:
-                    if example_flow.id not in flow_ids:
-                        flows.append(example_flow)
-            except Exception:  # noqa: BLE001
-                logger.exception("Error getting example flows")
+            stmt = stmt.where(Flow.is_component == True)  # noqa: E712
 
-        if remove_example_flows:
-            flows = [flow for flow in flows if flow.folder_id != folder.id]
+        if not get_all:
+            stmt = stmt.where(Flow.folder_id == folder_id)
+
+        if get_all:
+            flows = session.exec(stmt).all()
+            flows = validate_is_component(flows)
+            if components_only:
+                flows = [flow for flow in flows if flow.is_component]
+            if remove_example_flows and starter_folder_id:
+                flows = [flow for flow in flows if flow.folder_id != starter_folder_id]
+            if header_flows:
+                return [
+                    {"id": flow.id, "name": flow.name, "folder_id": flow.folder_id, "is_component": flow.is_component}
```

Comment:
Is this a non-backwards-compatible change? The response object used to include everything in the `FlowRead(FlowBase)` model, right?

Reply:
yes!

Reply:
Sure.

Previously, we were loading a large JSON containing all the flow data, even when only displaying the name and description. This meant the frontend had to store that entire flow in memory, including React flow information and everything required to render the components. Now, I'm retrieving a lightweight version of the flow to manage, and will only fetch the full data when it's needed to render the canvas.